src/upload-tests.proj

```diff
@@ -61,6 +61,8 @@
 
     <HelixJobType Condition="'$(HelixJobType)'==''">test/functional/cli/</HelixJobType>
 
+    <!-- Detect whether we are on a product construction build via ProductBuildId. If so, set source appropriately -->
+    <HelixSource Condition="'$(HelixSource)'=='' And '$(IsOfficial)'!='' And '$(TestProduct)'!='' And '$(Branch)'!='' And '$(ProductBuildId)'!=''">prodcon/$(TestProduct)/$(Branch)/</HelixSource>
```

Comment:
Couple points:
1. You can probably provide this externally from PipeBuild, so that it gets set properly for CoreCLR as well.
2. Why is ProductBuildId in the condition if it's not used in the constructed property?

Reply:
1. That's one way...we could just feed through an explicit HelixSource
2. It's simply to tell whether this is a prodcon build or not.